Mathlib/MeasureTheory/Measure/Regular.lean

```diff
@@ -13,42 +14,77 @@ import Mathlib.MeasureTheory.Constructions.BorelSpace.Basic
 A measure is `OuterRegular` if the measure of any measurable set `A` is the infimum of `μ U` over
 all open sets `U` containing `A`.
 
-A measure is `Regular` if it satisfies the following properties:
-* it is finite on compact sets;
-* it is outer regular;
-* it is inner regular for open sets with respect to compacts sets: the measure of any open set `U`
-  is the supremum of `μ K` over all compact sets `K` contained in `U`.
-
 A measure is `WeaklyRegular` if it satisfies the following properties:
 * it is outer regular;
 * it is inner regular for open sets with respect to closed sets: the measure of any open set `U`
   is the supremum of `μ F` over all closed sets `F` contained in `U`.
 
-In a Hausdorff topological space, regularity implies weak regularity. These three conditions are
-registered as typeclasses for a measure `μ`, and this implication is recorded as an instance.
-
-In order to avoid code duplication, we also define a measure `μ` to be `InnerRegular` for sets
+A measure is `Regular` if it satisfies the following properties:
+* it is finite on compact sets;
+* it is outer regular;
+* it is inner regular for open sets with respect to compacts closed sets: the measure of any open
+  set `U` is the supremum of `μ K` over all compact sets `K` contained in `U`.
+
+A measure is `InnerRegular` if it is inner regular for measurable sets with respect to compact
+closed sets: the measure of any measurable set `s` is the supremum of `μ K` over all compact
+sets contained in `s`.
```

Comment:
This docstring mentions compact closed sets, but the definition of InnerRegular is only about compact sets (not necessarily closed). I am curious about this because in the proof of Kolmogorov's extension theorem, we realized with Peter Pfaffelhuber that if we did not want to make assumptions like T2space, what we needed was regularity with respect to compact and closed sets (we obtain these compact closed sets as closures of compact sets).
Related to that, you also recently added `ClosableCompactSubsetOpenSpace` and it looks like the intended use of your new definitions in our case would be to use InnerRegular as you define it w.r.t. compact sets (not necessarily closed), then prove that the assumptions we have on our space imply `ClosableCompactSubsetOpenSpace` and use `exists_compact_closed_between`. Does that sound right?

Reply:
It's exactly that. The natural inner regularity assumption would be with respect to compact closed sets, and I started a refactor like that, but then I thought it would be too big a departure from existing literature, so I switched back only to compact sets (and forgot to adjust the docstring here). In the main application, to locally compact topological groups, it doesn't make a difference, because the topology is then automatically regular: this implies `ClosableCompactSubsetOpenSpace`, so regularity with respect to compact sets, or with respect to compact closed sets, is equivalent.

Reply:
Thanks for the clarification. I was also thinking about trying a closed-and-compact refactor for inner regular, but it was a daunting prospect. I'll review this pr to understand better how the various definitions you introduce interact.